### DISCUSSION TITLE
[SOLVED] The 4.0.0 is coming, or database transfer

### DISCUSSION THREAD

**Discussion by sakkamade** [martinrotter/rssguard#419] [Q&A] [answered]
Taking into consideration this issue #361, for novice user, there is not need whatever to use the 3.9.2, inasmuch as the databases of 3.9.2 and 4.0.0 shall be incompatible? 
Or the option of conversion of the database to version 4.0.0 will be implemented?

Could someone kindly elaborate on this point?

**Comment by martinrotter** [comment, marked as answer]
Hello @sakkamade.

Upcoming 4.0.0 release is probably the biggest set of changes ever made for RSS Guard, but majority of changes are done under the hood and are not directly visible for user - better DB schema, simple logic, easier to manage and read source code, some speed enhancements, some UX enhancements, many new features.

At this point I can say this for 3.9.2 and 4.0.0:
- configuration `ini` files are 100% compatible for both releases
- database `db` files are not compatible

As for the conversion script for the database. I am really really not sure if I will do it because the conversion will be relatively big one. So at this point once 4.0.0 (in one two months) I do not expect it to convert 3.9.x database into 4.x format.

I understand that this is a bit of a hassle for users, but it will be greatly benefical in the long run. Also, let me kindly emphasise here that whole RSS Guard application is free, open source, I have never made it for profit (I actually obtained like 80 USD in grand total in donations so far) and it is actually a one man show - almost nobody contributed source code.

So that is 3900 commits, 10 years of development, 50 000 lines of code. RSS Guard is nowhere near to be perfect RSS application. It sucks in many aspects, but 4.0.0 release will bring us much close to sustainable, more robust codebase. :)
